src/bio2bel/sources/intact.py

```diff
@@ -0,0 +1,532 @@
+            if i.startswith(prefix):
+                row_list.append(i)
+                flag = True
+        if not flag:
+            row_list.append(f'no {prefix} id')
+        final_list.append(row_list)
+    return final_list
+
+
+def get_processed_intact_df() -> pd.DataFrame:
+    """Load, filter and rename intact dataframe.
+
+    :return: processed dataframe
+    """
+    # original intact dataframe
+    df = _get_my_df()
+
+    # rename columns
+    df = rename_columns(df=df, columns_mapping=COLUMNS_MAPPING)
+
+    # take relevant columns for source, target, relation and PubMed ID
+    df = df[[SOURCE, TARGET, RELATION, PUBMED_ID]]
+
+    # drop nan value rows for interactor B
+    df = df[df[TARGET] != '-']
+
+    # filter for uniprot ids
+    df = df[df[SOURCE].str.contains("uniprot")]
+
+    # filter for pubmed
+    df[PUBMED_ID] = filter_for_prefix_multi(
+        list_ids=df[PUBMED_ID],
+        prefix='pubmed',
+    )
+
+    # filter interaction types
+    df[RELATION] = filter_for_prefix_single(
+        list_ids=df[RELATION],
+        rstrip=')',
+        lstrip='(',
+        separator='"',
+        prefix='(',
+    )
+
+    return df
+
+
+def get_bel() -> BELGraph:
+    """Get BEL graph.
+
+    :return: BEL graph
+    """
+    df = get_processed_intact_df()
+    graph = BELGraph(name=MODULE_NAME)
+    for _, row in tqdm(df.iterrows(), total=len(df.index), desc=f'mapping {MODULE_NAME}'):
+        _add_my_row(graph, row)
+    return graph
+
+
+def _add_my_row(graph: BELGraph, row) -> None:  # noqa:C901
+    """Add for every pubmed ID an edge with information about relationship type, source and target.
+
+    :param graph: graph to add edges to
+    :param row: row with metainformation about source, target, relation, pubmed_id
+    :return: None
+    """
+    relation = row[RELATION]
+    source_uniprot_id = row[SOURCE]
+    target_uniprot_id = row[TARGET]
+
+    pubmed_ids = row[PUBMED_ID]
+
+    source = pybel.dsl.Protein(
+        namespace='uniprot',
+        identifier=source_uniprot_id,
+        name=get_mnemonic(source_uniprot_id),
+    )
+    target = pybel.dsl.Protein(
+        namespace='uniprot',
+        identifier=target_uniprot_id,
+        name=get_mnemonic(target_uniprot_id),
+    )
+
+    for pubmed_id in pubmed_ids:
+
+        # INCREASE
+
+        if relation in INTACT_INCREASES_ACTIONS:
+
+            # oxidoreductase activity
+            if relation == 'oxidoreductase activity electron transfer reaction':
+                target_mod = target.with_variants(
+                    pybel.dsl.ProteinModification('Red'),
+                )
+
+                graph.add_increases(
+                    source,
+                    target_mod,
+                    citation=pubmed_id,
+                    evidence=EVIDENCE,
+                    subject_modifier=pybel.dsl.activity(),
+                )
+                continue
+            # isomerase reaction
+            elif relation == 'isomerase reaction' \
```

Comment:
don't make multiple lines of equals checks, you can do something like `relation in {'A', 'B' ...}:`

Reply:
Resolved

Reply:
0ec88265d7b3ff17bf909655b9f826ea873418d3